src/components/EditorHeader/EditorHeader.vue

```diff
@@ -0,0 +1,218 @@
+import SbMobileViwer from './components/MobileViwer'
+import SbHeaderTitle from './components/HeaderTitle'
+
+import SbHeaderItem from './components/HeaderItem'
+import SbIcon from '../Icon'
+
+import SbButton from '../Button'
+
+import { SbMenu, SbMenuButton, SbMenuList, SbMenuItem } from '../Menu'
+
+import { sharedProps } from './lib'
+
+export default {
+  name: 'SbEditorHeader',
+
+  components: {
+    SbDesktopAndTabletViwer,
+    SbHeaderTitle,
+    SbMobileViwer,
+    SbButton,
+    SbHeaderItem,
+    SbIcon,
+    SbMenu,
+    SbMenuButton,
+    SbMenuList,
+    SbMenuItem,
+  },
+
+  props: {
+    ...sharedProps,
+  },
+
+  data: () => ({
+    window: {
+      width: 0,
+    },
+  }),
+
+  computed: {
+    returnNameOfIconStatus() {
+      return this.spaceStatus === 'published'
+        ? 'published-circle'
+        : 'status-circle'
+    },
+
+    returnColorOfStatus() {
+      return this.spaceStatus === 'unpublished'
+        ? 'light'
+        : this.spaceStatus === 'published'
+        ? 'primary'
+        : this.spaceStatus === 'published-and-changes'
+        ? 'light-gray'
+        : 'light'
+    },
+
+    returnViewFormat() {
+      return this.window.width > 880
+        ? 'desktop'
+        : this.window.width > 600 && this.window.width < 880
+        ? 'tablet'
+        : 'mobile'
+    },
+  },
+
+  created() {
+    window.addEventListener('resize', this.$_handleResize)
+    this.$_handleResize()
+  },
+
+  destroyed() {
+    window.removeEventListener('resize', this.$_handleResize)
+  },
+
+  methods: {
+    /**
+     * method to listen and chenge the window.width
+     */
+    $_handleResize() {
+      this.window.width = window.innerWidth
+    },
+    /**
+     * this method is triggered when an action is taken on the
+     * desktop and mobile components, it works as an orchestrator,
+     * to distribute the emits.
+     */
+    $_handleListenersOrchestrator(action) {
+      if (action.type === 'lang') {
+        this.handleSetNewLanguage(action.language)
+      }
+      if (action.type === 'act') {
+        this.handleSelectAction(action.action)
+      }
+    },
+    /**
+     * method to emit the selection of new option
+     */
+    handleSelectOptions(opt) {
+      this.$emit('select-options', opt.toLowerCase().replace(' ', '-'))
+    },
+    /**
+     * method to emit save changes on space
+     */
+    handleSaveChanges() {
+      this.$emit('save')
+    },
+    /**
+     * method to emit the publish event
+     */
+    handlePublishSpace() {
+      this.$emit('publish')
+    },
+    /**
+     * method to emit the new action
+     */
+    handleSelectAction(act) {
+      this.$emit('select-action', act.toLowerCase().replace(' ', '-'))
+    },
+    /**
+     * method to emit the selection of a language
+     */
+    handleSetNewLanguage(lang) {
+      this.$emit('change-language', { language: lang })
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+@import './editor-header.scss';
```

Comment:
Are we importing the styles inside the component?

Reply:
My mistake, I forgot to remove